### ISSUE TITLE
NumberFormatException on getting the connection

### ISSUE THREAD

**Issue by axsy** [impossibl/pgjdbc-ng#472]
While trying to connect to my database I'm getting `NumberFormatException` with the following stacktrace:

```
Caused by: java.lang.NumberFormatException: For input string: "2222793757"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:583)
	at java.lang.Integer.valueOf(Integer.java:766)
	at com.impossibl.postgres.system.procs.Int4s$TxtDecoder.decodeNativeValue(Int4s.java:110)
	at com.impossibl.postgres.system.procs.Int4s$TxtDecoder.decodeNativeValue(Int4s.java:97)
	at com.impossibl.postgres.system.procs.AutoConvertingTextDecoder.decodeValue(BaseDecoders.java:277)
	at com.impossibl.postgres.system.procs.BaseTextDecoder.decode(BaseDecoders.java:216)
	at com.impossibl.postgres.system.procs.BaseTextDecoder.decode(BaseDecoders.java:194)
	at com.impossibl.postgres.protocol.v30.BufferRowData.getField(BufferRowData.java:145)
	at com.impossibl.postgres.system.tables.Table.getFieldOfRow(Table.java:86)
	at com.impossibl.postgres.system.tables.PGTypeTable$Row.load(PGTypeTable.java:98)
	at com.impossibl.postgres.system.tables.PGTypeTable.createRow(PGTypeTable.java:272)
	at com.impossibl.postgres.system.tables.PGTypeTable.createRow(PGTypeTable.java:46)
	at com.impossibl.postgres.system.tables.Tables.convertRows(Tables.java:83)
	at com.impossibl.postgres.system.tables.Table.query(Table.java:79)
	at com.impossibl.postgres.system.BasicContext.lambda$loadTypes$1(BasicContext.java:423)
	at com.impossibl.postgres.types.SharedRegistry.seed(SharedRegistry.java:291)
	at com.impossibl.postgres.system.BasicContext.loadTypes(BasicContext.java:471)
	at com.impossibl.postgres.system.BasicContext.init(BasicContext.java:378)
	at com.impossibl.postgres.jdbc.PGDirectConnection.init(PGDirectConnection.java:285)
	at com.impossibl.postgres.jdbc.ConnectionUtil.createConnection(ConnectionUtil.java:311)
	at com.impossibl.postgres.jdbc.ConnectionUtil.createConnection(ConnectionUtil.java:271)
	at com.impossibl.postgres.jdbc.PGDriver.connect(PGDriver.java:119)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:208)
```

It happened because `pg_catalog.pg_type` returned the type with `oid` 2222793757 which is treated by driver as signed 32-bit integer being an unsigned 32-bit integer instead (https://www.postgresql.org/docs/9.6/datatype-oid.html).

An issue is reproducible with the `0.8.3` and latest `master`.

**Comment by kdubb** [comment]
@axsy Can you verify #473 works for you?

**Comment by kdubb** [comment]
This is now available in `0.8.4-SNAPSHOT` and will be released with `0.8.4`. If the PR has not fixed your issue please re-open.
